control-plane/api-gateway/common/resources.go

```diff
@@ -364,6 +365,25 @@ func (s *ResourceMap) ReferenceCountHTTPRoute(route gwv1beta1.HTTPRoute) {
 	s.httpRouteGateways[consulKey] = set
 }
 
+func (s *ResourceMap) AddHTTPRouteExternalFilter(route gwv1beta1.HTTPRoute, filter client.Object) {
+	key := client.ObjectKeyFromObject(&route)
+	consulKey := NormalizeMeta(s.toConsulReference(api.HTTPRoute, key))
+
+	set, ok := s.httpRouteGateways[consulKey]
+	if !ok {
+		s.ReferenceCountHTTPRoute(route)
+		set = s.httpRouteGateways[consulKey]
+	}
+	if set.externalFilters == nil {
+		//initialize
+		set.externalFilters = mapset.NewSet()
+	}
+
+	set.externalFilters.Add(filter)
```

Comment:
Would it make sense to just store this as a map of keys --> kubernetes objects? Also why make this a `client.Object` rather than just passing the CRD type directly? Ideally if we have a bad filter, the whole route is considered invalid and not bound, so no need to store anything overloaded here other than the CRD objects themselves (I believe that mapset also requires object hashing which I'm not sure that `client.Object` will actually implement, so this code might panic).